sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/Credential/DataSourceCredentialEntity.cs

```diff
@@ -76,13 +77,68 @@ internal DataSourceCredentialPatch GetPatchModel()
                 {
                     Parameters = new() { ConnectionString = c.ConnectionString }
                 },
-                _ => throw new InvalidOperationException("Invalid data source credential type")
+                _ => new DataSourceCredentialPatch()
             };
 
+            patch.DataSourceCredentialType = DataSourceCredentialType;
             patch.DataSourceCredentialName = Name;
             patch.DataSourceCredentialDescription = Description;
 
             return patch;
         }
+
+        internal static DataSourceCredentialEntity DeserializeDataSourceCredentialEntity(JsonElement element)
```

Comment:
is there an easy way to know if there are improvements into codegen on how we do deserialization, to know and update this?

Reply:
Discussed offline:
> Regarding codegen improvements, I think currently there's not much we can do other than be attentive. Regarding new properties being added, we use the internal base constructor that takes all properties, even optional ones. If a new property is added and we regenerate the code, we'll get a compilation error.